Subject: acpi video device enumeration: fix incorrect device list allocation
Date:	Sun, 02 Jan 2005 19:45:13 +0000
X-BK-ChangeSetKey: torvalds@evo.osdl.org|ChangeSet|20050102194513|01534
Patch-mainline: yes
From: olh@suse.de


ChangeSet 1.2131, 2005/01/02 11:45:13-08:00, torvalds@evo.osdl.org

	acpi video device enumeration: fix incorrect device list allocation
	
	It didn't allocate space for the final terminating entry,
	which caused it to overwrite the next slab entry, which in turn
	sometimes ended up being a slab array cache pointer. End result:
	total slab cache corruption at a random time afterwards. Very
	nasty.



 video.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)


diff -Nru a/drivers/acpi/video.c b/drivers/acpi/video.c
--- a/drivers/acpi/video.c	2005-01-02 13:05:53 -08:00
+++ b/drivers/acpi/video.c	2005-01-02 13:05:53 -08:00
@@ -1524,7 +1524,7 @@
 		dod->package.count));
 
 	active_device_list= kmalloc(
- 		dod->package.count*sizeof(struct acpi_video_enumerated_device),
+ 		(1+dod->package.count)*sizeof(struct acpi_video_enumerated_device),
 	       	GFP_KERNEL);
 
 	if (!active_device_list) {
-
To unsubscribe from this list: send the line "unsubscribe bk-commits-head" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html

